src/Oscar.jl

```diff
@@ -404,38 +404,60 @@ end
 @doc Markdown.doc"""
     test_module(x, new::Bool = true)
 
-Run the Oscar tests in the file `x.jl` where `x` is a string.
+Run the Oscar tests in the file `test/x.jl` where `x` is a string.
 
 If `x == "all"` run the entire testsuite.
 
 The optional parameter `new` takes the values `false` and `true` (default). If
 `true`, then the tests are run in a new session, otherwise the currently active
 session is used.
 """
-function test_module(x::AbstractString, new::Bool = true)
-   julia_exe = Base.julia_cmd()
-   if x == "all"
-     test_file = joinpath(oscardir, "test/runtests.jl")
-   else
-     test_file = joinpath(oscardir, "test/$x.jl")
-   end
+function test_module(x::AbstractString, new::Bool=true)
+  julia_exe = Base.julia_cmd()
+  if x == "all"
+    test_file = joinpath(oscardir, "test/runtests.jl")
+  else
+    test_file = joinpath(oscardir, "test/$x.jl")
+  end
 
-   if new
-     cmd = "using Test; using Oscar; Hecke.assertions(true); include(\"$test_file\");"
-     @info("spawning ", `$julia_exe -e \"$cmd\"`)
-     run(`$julia_exe -e $cmd`)
-   else
-     @info("Running tests for $x in same session")
-     try
-       include(test_file)
-     catch e
-       if isa(e, LoadError)
-         println("You need to do \"using Test\"")
-       else
-         rethrow(e)
-       end
-     end
-   end
+  if new
+    cmd = "using Test; using Oscar; Hecke.assertions(true); include(\"$test_file\");"
+    @info("spawning ", `$julia_exe -e \"$cmd\"`)
+    run(`$julia_exe -e $cmd`)
+  else
+    @info("Running tests for $x in same session")
+    try
+      include(test_file)
+    catch e
+      if isa(e, LoadError)
+        println("You need to do \"using Test\"")
+      else
+        rethrow(e)
+      end
+    end
+  end
+end
+
+@doc Markdown.doc"""
+    test_experimental_module(project::AbstractString, x::AbstractString="runtests", new::Bool = true)
+
+Run the Oscar tests in the file `experimental/project/test/x.jl`.
+
+If `x == "all"` run the entire testsuite of the project `project`.
+
+The optional parameter `new` takes the values `false` and `true` (default). If
+`true`, then the tests are run in a new session, otherwise the currently active
+session is used.
+"""
+function test_experimental_module(
+  project::AbstractString, x::AbstractString, new::Bool=true
+)
+  if x == "all"
+    test_file = "../experimental/$project/test/runtests"
+  else
+    test_file = "../experimental/$project/test/$x"
+  end
+  test_module(test_file, new)
```

Comment:
I am not really sure, what the convention is, but one alternative could be to put a `;` behind `project::...` to mark the other two parameters as optional. Also I'd prefer a better name for `x`, maybe `testfile` or `testscript` to keep it short. What do you think?

Reply:
I was thinking about that as well. Then `x::AbstractString="runtests"` would be easier to understand than the default `"all"`. To keep consistency with `test_module`, I would then do the same there, i.e. making the path `x` and `new` kwargs. To keep the name short (it is a kwarg now), I would suggest renaming `x` to `file` in `test_module` and `test_experimental_module`.
Is that what you had in mind?

Reply:
Funny, I was just wondering, who named it `x` in `test_module`, and git blame give:
```
50176b21c43 (Lars Kastner           2022-09-28 10:20:44 +0200 414) function test_module(x::AbstractString, new::Bool = true)
```
Joke aside:
I agree that consistency is a great thing, and to rename there as well. Do we need a deprecation then? I'd say no, since this does not affect functionality of Oscar, but is a developer tool.